### DISCUSSION TITLE
set application name postgres before commit m how i can make?

### DISCUSSION THREAD

**Discussion by johndiego** [sqlalchemy/sqlalchemy#9707] [Usage Questions] [answered]
I'm creating an audit trigger in the database, and I would like to know if there's any way to set the application_name before performing a commit (INSERT ,UPDATE,DELETE), I've tried before_insert before_update, I'd like to do this before any transaction, I've tried several times things and I've been having this problem for a week, I'm sorry for my English.
thank you very much.
i 'm using 
**Flask-SQLAlchemy==2.4.1**
**SQLAlchemy==1.3.5**
I'm read this page.
https://docs.sqlalchemy.org/en/13/core/events.html#sqlalchemy.events.ConnectionEvents.commit

![image](https://user-images.githubusercontent.com/18476845/234805919-18af203c-3e8c-47c4-8033-f407ebb6e0fd.png)
**is not fired..**

**Comment by lelit** [comment, marked as answer]
With a similar goal, I hook my logic to the `begin` event on the engine, something like
```python
def get_engine(settings):
    "Setup the SA engine taking details from environment variables."

    engine = create_sqlalchemy_engine(verbose=asbool(settings.get('sqlalchemy.echo')))

    @event.listens_for(engine, 'begin')
    def set_user_context(conn):
        "Inject the current user info into the transaction."

        from .i18n import DEFAULT_LOCALE

        request = get_current_request()
        if request is not None:
            conn.execute(
                text("SELECT auth.set_current_user_id_and_locale(:uid, :ul, :dl)"),
                uid=request.unauthenticated_userid,
                ul=request.locale_name,
                dl=getattr(request, 'DEFAULT_LOCALE', DEFAULT_LOCALE))

    return engine
```

where the `set_current_user_id_and_locale()` injects a few _custom variables_ into the PostgreSQL execution context using its `set_config()` function, perused by subsequent triggers.

**Reply by johndiego** [reply]
thanks for reply , 
```python
import sqlalchemy
from flask import Flask,current_app
from sqlalchemy import event,inspect,PickleType, cast,create_engine
from flask_sqlalchemy import SQLAlchemy
db = SQLAlchemy()
```
how i can implement this hook in aplication?

**Reply by lelit** [reply]
I do not use `Flask`, but a quick glance at the [documentation](https://flask-sqlalchemy.palletsprojects.com/en/3.0.x/quickstart/#what-to-remember) seems to suggest that you can reach the engine as `db.engine` (maybe after calling `db.init_app()`?).

If so, maybe

```python
@event.listens_for(db.engine, 'begin')
def set_application_name(conn):
    ...
```

does the job?

**Reply by johndiego** [reply]
I will try.

**Comment by johndiego** [comment]
@lelit veryyyyyyyy Thanks!!
